src/peergos/shared/user/Groups.java

```diff
@@ -0,0 +1,52 @@
+package peergos.shared.user;
+
+import peergos.shared.cbor.*;
+import peergos.shared.crypto.random.*;
+import peergos.shared.util.*;
+
+import java.util.*;
+import java.util.function.*;
+
+public class Groups implements Cborable {
+    public final Map<String, String> uidToGroupName;
+
+    public Groups(Map<String, String> uidToGroupName) {
+        this.uidToGroupName = uidToGroupName;
+    }
+
+    public static Groups generate(SafeRandom r) {
+        Map<String, String> uidToNames = new TreeMap<>();
+        uidToNames.put(generateUid(r), SocialState.FRIENDS_GROUP_NAME);
+        uidToNames.put(generateUid(r), SocialState.FOLLOWERS_GROUP_NAME);
+        return new Groups(uidToNames);
+    }
+
+    /* Generate a uid that cannot clash with a username, but which is a valid filename
+     */
+    public static String generateUid(SafeRandom r) {
+        return "." + ArrayOps.bytesToHex(r.randomBytes(32));
```

Comment:
maybe also add a prefix lilke "." + "G_" + random. In case we need to add another set of files for some other purpose in the future

Reply:
if we need to add other we can do that safely without changing groups. e.g. with a non hex character after the .